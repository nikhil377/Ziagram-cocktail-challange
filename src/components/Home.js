import React from 'react'
import axios from 'axios'

import CocktailIndex from './CocktailIndex'
import CocktailTitle from './CocktailTitle'

class Home extends React.Component {
  constructor(){
    super()

    this.state = {
      search: [],
      data: null,
      filter: 'ingredient',
      placeholder: 'search for ordinary_drink'
    }

    this.handleChange = this.handleChange.bind(this)
    this.handleSubmit = this.handleSubmit.bind(this)
    this.handleFilterChange = this.handleFilterChange.bind(this)
  }

  handleChange(e){
    const searchData = ({ ...this.state.search, [e.target.name]: e.target.value })
    this.setState({search: searchData})
  }

  handleSubmit(e) {
    e.preventDefault()
    console.log("filter", this.state.filter)
    const endpoint = this.state.filter === 'ingredient' ? 'filter.php?i' :this.state.filter === 'name' ? 'search.php?s': 'filter.php?c'
    axios.get(`https://www.thecocktaildb.com/api/json/v1/1/${endpoint}=${this.state.search.searchInput}`)
      .then(res => this.setState({ data: res.data }))
      .then(() => this.searchResultsSection.scrollIntoView({ behavior: 'smooth', block: 'start' }))
  }

  handleFilterChange(e){
    const filter = e.target.value
    this.setState({filter: filter},()=>{
      this.state.filter === 'ingredient'? this.setState({placeholder: 'search for vodka or wine'}): 
      this.state.filter === 'name'? this.setState({placeholder: 'search for vodka or wine'}): 
      this.setState({placeholder:'search for ordinary_drink'})
    })
  }


  render() {
    return(
      <section>
        <CocktailTitle />
        <div className="container">
          <section className="section search-results" ref={elem => this.searchResultsSection = elem}>
            <form  className="search-form" onSubmit={this.handleSubmit}>
              <div className="field">
                <div className="control" onChange={this.handleFilterChange}>
                <label className="radio">
                    <input type="radio" name="searchFilter" value="category" defaultChecked={true}  /> category
                  </label>
                  <label className="radio">
                    <input type="radio" name="searchFilter" value="name" /> name
                  </label>
                  <label className="radio">
                    <input type="radio" name="searchFilter" value="ingredient"/> ingredient
                  </label>
                </div>
              </div>
              <div className="field has-addons">
                <div className="control is-expanded">
                  <input
                    id="input"
                    className="input"
                    type="text"
                    name="searchInput"
                    placeholder={this.state.placeholder}
                    onChange={this.handleChange}
                  />
                </div>
                <div className="control">
                  <button className="button search-button">Search</button>
                </div>
              </div>
            </form>
          </section>
          <section className="section">
            <CocktailIndex {...this.state.data}/>
          </section>
        </div>
      </section>
    )
  }

}

export default Home
