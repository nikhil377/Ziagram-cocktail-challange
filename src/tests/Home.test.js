// MyComponent.test.js
import React from 'react';
import Home from '../components/Home';
import { configure ,shallow} from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';

configure({ adapter: new Adapter() });
let wrapper;
beforeEach(()=>{
    wrapper = shallow(<Home />);
})
describe("Homepage testcase", () => {
  test('should render the homepage', () => {
      expect(wrapper.getElements()).toMatchSnapshot();
});
test('it should render the search button of homepage',()=>{
    expect(wrapper.find('button').text()).toBe('Search');
})
test('it should render the input search of search bar ',()=>{
    expect(wrapper.find('#input').text()).toBe('');
})
});
  
