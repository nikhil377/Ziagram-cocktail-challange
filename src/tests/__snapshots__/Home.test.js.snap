// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Homepage testcase should render the homepage 1`] = `
Array [
  <section>
    <CocktailTitle />
    <div
      className="container"
    >
      <section
        className="section search-results"
      >
        <form
          className="search-form"
          onSubmit={[Function]}
        >
          <div
            className="field"
          >
            <div
              className="control"
              onChange={[Function]}
            >
              <label
                className="radio"
              >
                <input
                  defaultChecked={true}
                  name="searchFilter"
                  type="radio"
                  value="category"
                />
                 category
              </label>
              <label
                className="radio"
              >
                <input
                  name="searchFilter"
                  type="radio"
                  value="name"
                />
                 name
              </label>
              <label
                className="radio"
              >
                <input
                  name="searchFilter"
                  type="radio"
                  value="ingredient"
                />
                 ingredient
              </label>
            </div>
          </div>
          <div
            className="field has-addons"
          >
            <div
              className="control is-expanded"
            >
              <input
                className="input"
                id="input"
                name="searchInput"
                onChange={[Function]}
                placeholder="search for ordinary_drink"
                type="text"
              />
            </div>
            <div
              className="control"
            >
              <button
                className="button search-button"
              >
                Search
              </button>
            </div>
          </div>
        </form>
      </section>
      <section
        className="section"
      >
        <CocktailIndex />
      </section>
    </div>
  </section>,
]
`;
